lib/from_openstudio/model.rb

```diff
@@ -57,6 +61,7 @@ def self.translate_from_openstudio(openstudio_model)
       hash[:tolerance] = 0.01
       hash[:angle_tolerance] = 1.0
 
+      $shade_construction = [] # Array for all shade construction
       hash[:properties] = properties_from_model(openstudio_model)
```

Comment:
I realised that the construction is being created before other shade constructions from orphaned shade, room, face, aperture shades can be added to the $shade_construction hash and the method is called.

Reply:
Is this a question? Or a note that this code does not work?

In any event, I agree that `$shade_construction` should be a hash as you say and not a list as it is currently written in the code.